src/main/webapp/WEB-INF/pages/product.jsp

```diff
@@ -33,5 +37,23 @@
             ${product.stock}
         </td>
       </tr>
+      <tr>
+        <td>
+            <button type="submit" name="buttonAdd2Cart">Add to cart</button>
```

Comment:
Please add quantity input field to indicate the number of product to add to the cart + add validation: it should be a number, > 0, etc.